plugins/mac-avcapture/OBSAVCapture.m

```diff
@@ -475,34 +419,64 @@ - (BOOL)configureSession:(NSError *__autoreleasing *)error
 
     CMTime time = {.value = fps.denominator, .timescale = fps.numerator, .flags = 1};
 
-    AVCaptureDeviceFormat *format = nil;
-
-    for (AVCaptureDeviceFormat *formatCandidate in [self.deviceInput.device.formats reverseObjectEnumerator]) {
-        CMVideoDimensions formatDimensions = CMVideoFormatDescriptionGetDimensions(formatCandidate.formatDescription);
+    const char *selectedFormat = obs_data_get_string(self.captureInfo->settings, "supported_format");
+    NSString *selectedFormatNSString = [NSString stringWithCString:selectedFormat encoding:NSUTF8StringEncoding];
```

Comment:
In theory you should be able to simply box the character array pointer here via `@(selectedFormat)`, but you need to explicitly check against a possible NULL pointer before boxing.

Reply:
Fixed.